src/cantools/subparsers/__utils__.py

```diff
@@ -135,12 +135,15 @@ def format_message_by_frame_id(dbase : Database,
         else:
             return f' Frame 0x{frame_id:x} is a container message'
 
-    return format_message(message,
-                          data,
-                          decode_choices,
-                          single_line,
-                          allow_truncated=allow_truncated,
-                          allow_excess=allow_excess)
+    try:
+        return format_message(message,
+                            data,
+                            decode_choices,
+                            single_line,
+                            allow_truncated=allow_truncated,
+                            allow_excess=allow_excess)
+    except Exception as e:
```

Comment:
catching all exceptions here is an awfully wide net. catching `DecodeError` is sufficient?